editor/src/components/canvas/canvas-strategies/strategies/resize-helpers.ts

```diff
@@ -99,16 +107,51 @@ export function getLockedAspectRatio(
   interactionData: InteractionSession,
   modifiers: Modifiers,
   originalBoundingBox: CanvasRectangle,
+  anySelectedElementAspectRatioLocked: boolean,
 ): number | null {
   if (interactionData.aspectRatioLock != null) {
     return interactionData.aspectRatioLock
   }
-  if (modifiers.shift) {
-    return originalBoundingBox.width / originalBoundingBox.height
+
+  if (anySelectedElementAspectRatioLocked && modifiers.shift) {
+    return null
+  }
+
+  if (anySelectedElementAspectRatioLocked || modifiers.shift) {
+    return originalBoundingBox.width / (originalBoundingBox.height || 1)
   }
+
   return null
 }
 
+function isElementImage(instance: ElementInstanceMetadata) {
+  return defaultEither(
+    false,
+    mapEither((e) => e.type === 'JSX_ELEMENT' && e.name.baseVariable === 'img', instance.element),
+  )
+}
+
+function isElementAspectRatioLocked(instance: ElementInstanceMetadata) {
+  return defaultEither(
+    false,
+    mapEither(
+      (e) => e.type === 'JSX_ELEMENT' && getJSXAttribute(e.props, AspectRatioLockedProp),
+      instance.element,
+    ),
+  )
+}
+
+export function isAnySelectedElementAspectRatioLocked(
+  jsxMetadata: ElementInstanceMetadataMap,
+  selectedElements: Array<ElementPath>,
+): boolean {
+  return MetadataUtils.findElementsByElementPath(jsxMetadata, selectedElements).some((e) => {
+    const h = isElementImage(e)
+    const w = isElementAspectRatioLocked(e)
+    return h || w
```

Comment:
h and w? w the f? :D

Reply:
I think `return isElementImage(e) || isElementAspectRatioLocked(e)` is better. 
no need to push temporary variables in the PR reader's mental stack only to pop them immediately!

Reply:
absolutely, that was the original version, the one that you commented on got pushed by accident